fix_: applying removed contacts from backup

saledjenic opened the pull request:
Applying removed contact from the backup was removed due to condition.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (9)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8485653a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/1/) | 2024-10-07 15:34:54 | ~3 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/1//consoleText) |
| :heavy_check_mark: | 8485653a | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5925/1/) | 2024-10-07 15:35:39 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241007-153059-848565-pr5925.zip) |
| :heavy_check_mark: | 8485653a | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5925/1/) | 2024-10-07 15:36:15 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241007-153056-848565-pr5925.aar) |
| :heavy_check_mark: | 8485653a | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/1/) | 2024-10-07 16:04:51 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | cf3f847e | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5925/2/) | 2024-10-07 19:37:44 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-241007-193532-cf3f84-pr5925.aar) |
| :heavy_check_mark: | cf3f847e | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-5925/2/) | 2024-10-07 19:38:16 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-241007-193535-cf3f84-pr5925.zip) |
| :heavy_check_mark: | cf3f847e | [#2](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/2/) | 2024-10-07 19:39:12 | ~3 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/2//consoleText) |
| :heavy_check_mark: | cf3f847e | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5925/2/) | 2024-10-07 19:39:25 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-241007-193533-cf3f84-pr5925.zip) |
| :heavy_check_mark: | cf3f847e | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/2/) | 2024-10-07 20:08:44 | ~33 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/2//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 0690be28 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5925/3/) | 2025-01-30 12:32:56 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250130-122907-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-5925/1/) | 2025-01-30 12:33:07 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250130-122912-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5925/3/) | 2025-01-30 12:34:16 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250130-122909-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-5925/1/) | 2025-01-30 12:34:33 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250130-122912-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-5925/1/) | 2025-01-30 12:34:45 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250130-122917-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5925/3/) | 2025-01-30 12:35:17 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250130-122906-0690be-pr5925.aar) |
| :heavy_multiplication_x: | 0690be28 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/3/) | 2025-01-30 12:35:48 | ~6 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/3//consoleText) |
| :heavy_check_mark: | 0690be28 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/3/) | 2025-01-30 13:00:04 | ~30 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/3//consoleText) |
| :heavy_check_mark: | 0690be28 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/main/job/PR-5925/2/) | 2025-02-10 15:34:26 | ~3 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250210-153033-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/4/) | 2025-02-10 15:34:56 | ~4 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc/job/PR-5925/4//consoleText) |
| :heavy_check_mark: | 0690be28 | [#2](https://ci.status.im/job/status-go/job/prs/job/macos/job/x86_64/job/main/job/PR-5925/2/) | 2025-02-10 15:35:52 | ~5 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-250210-153033-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#2](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/main/job/PR-5925/2/) | 2025-02-10 15:35:54 | ~5 min | `windows` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-windows-250210-153037-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/x86_64/job/main/job/PR-5925/4/) | 2025-02-10 15:36:00 | ~5 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-linux-250210-153031-0690be-pr5925.zip) |
| :heavy_check_mark: | 0690be28 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-5925/4/) | 2025-02-10 15:36:34 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-250210-153029-0690be-pr5925.aar) |
| :heavy_check_mark: | 0690be28 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/4/) | 2025-02-10 16:02:12 | ~31 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5925/4//consoleText) |
| :heavy_check_mark: | 0690be28 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-5925/4/) | 2025-02-10 18:54:07 | ~3 hr 23 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-250210-153030-0690be-pr5925.zip) |
| :x: | 0690be28 | [#1](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/nwaku/job/PR-5925/1/) | 2025-12-01 11:27:41 | ~1 min | `windows` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/windows/job/x86_64/job/nwaku/job/PR-5925/1/consoleText) |
| :heavy_check_mark: | 0690be28 | [#1](https://ci.status.im/job/status-go/job/prs/job/macos/job/aarch64/job/nwaku/job/PR-5925/1/) | 2025-12-01 11:35:34 | ~24 min | `macos` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-macos-251201-111035-0690be-pr5925.zip) |
| :heavy_multiplication_x: | 0690be28 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-nwaku/job/PR-5925/1/) | 2025-12-01 12:51:25 | ~38 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-nwaku/job/PR-5925/1//consoleText) |
| :heavy_multiplication_x: | 0690be28 | [#1](https://ci.status.im/job/status-go/job/prs/job/tests-rpc-nwaku/job/PR-5925/1/) | 2025-12-01 13:27:47 | ~58 min | `tests-rpc` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests-rpc-nwaku/job/PR-5925/1//consoleText) |

jrainville commented on protocol/messenger_handler.go:
```diff
@@ -583,6 +583,8 @@ func (m *Messenger) handleSyncChats(messageState *ReceivedMessageState, chats []
 func (m *Messenger) HandleSyncInstallationContactV2(state *ReceivedMessageState, message *protobuf.SyncInstallationContactV2, statusMessage *v1protocol.StatusMessage) error {
 	// Ignore own contact installation
```
This comment is no longer on the right line

saledjenic replied:
Sorry, was testing something else and that left. Removed now.

jrainville commented on protocol/messenger_handler.go:
```diff
@@ -583,6 +583,8 @@ func (m *Messenger) handleSyncChats(messageState *ReceivedMessageState, chats []
 func (m *Messenger) HandleSyncInstallationContactV2(state *ReceivedMessageState, message *protobuf.SyncInstallationContactV2, statusMessage *v1protocol.StatusMessage) error {
 	// Ignore own contact installation
 
+	fromBackup := statusMessage == nil
```
It's not clear why having no message means it comes from backups :thinking:

saledjenic replied:
Well, there are 3 places where this function is called from:
1. `handleBackup` (used when applying messages from the backup)
2. `HandleSyncRawMessages` (used when local pairing)
3. `dispatchToHandler` -> `handleSyncInstallationContactV2Protobuf` (used when a synced message comes from another (paired) device over the network)

In case of point 3 the code is the same as it was, in case of points 1 and 2 we need to store the contact even it has `Removed` or `Blocked` flag set to `false`.

That's why I used this flag `fromPairedDevice := statusMessage != nil` to filter out the source from which messages are coming.

So the issue was spotted in https://github.com/status-im/status-mobile/issues/18096 but based on the code the same happens for Removed or Blocked contacts for applying messages from backup and local pairing. That should be fixed now.

saledjenic committed:
fix_: store contacts from the backup that were removed

osmaczko reviewed (commented):
Thanks. Could you please elaborate on how this resolves the issue of a deleted contact being restored from backup when it shouldn't? From what I understand, the additional condition enforces the creation of a removed contact in all scenarios except for network syncing. Additionally, a regression test would be extremely helpful.

saledjenic commented:
@osmaczko have you tested? Are you able to reproduce the issue with these changes?

> From what I understand, the additional condition enforces the creation of a removed contact in all scenarios except for network syncing.

Actually that's the thing. When we do a backup, we send all contacts from the db, including those marked as removed/blocked. When we restore from waku or from another device, we actually want to replicate the db as it was when the last backup was done. When a removed contact is about to be added, since it's not in db, thus not found, we just "return" instead of adding it to db. What I did, I let the function "return" if the sync message is coming from paired device over a network, but let it be added to db in case it's coming from restoring from waku or another device.

churik commented:
@saledjenic 
Sorry for extremely late notice, does it make sense to check it now? we need to rebase then status-go and we'll able to re-check the issue again. Seems the issue got lost in other priorities.

saledjenic commented:
@churik since this PR is not merged yet, I guess that the issue is still here, you can test the app with and without this change and see. Do you want me to rebase this branch?

churik commented:
@saledjenic yes, please

pavloburykh commented:
@saledjenic thanks for the PR. Please take a look at the issues logged in corresponding mobile PR https://github.com/status-im/status-mobile/pull/22011#issuecomment-2631006998

pavloburykh commented:
> @saledjenic thanks for the PR. Please take a look at the issues logged in corresponding mobile PR [status-im/status-mobile#22011 (comment)](https://github.com/status-im/status-mobile/pull/22011#issuecomment-2631006998)

Hey @saledjenic! Pinging you in case you missed the comment. Thanks.

saledjenic commented:
@pavloburykh when I made this change, it was applicable and was fixing the issue, but since then we've changed the onboarding flow a lot, changing the order and the logic of doing things. At the same time, I am not sure that we've updated all handlers, basically the entire logic, on the statusgo side to handle all those changes, cause the logic from that time relied on the request that after up to 2 mins spent on the fetching data from waku the app discards all future received messages.

Also I posted a comment to your other PR where you've commented about the issue. Please check here https://github.com/status-im/status-mobile/pull/22011#issuecomment-2647966562

pavloburykh commented:
Thanks  @saledjenic. I am closing this mobile PR for now, as we probably will need another PR after mobile core will take a look. Let's for now continue and keep discussion in one place, for example in the [initial issue](https://github.com/status-im/status-mobile/issues/18096). I have left the links to our discussion [there](https://github.com/status-im/status-mobile/issues/18096#issuecomment-2648378330).

igor-sirotin commented:
@saledjenic is this still relevant? Should we finish it or close?

saledjenic commented:
@igor-sirotin it's more of a question for the backend team. At that time, that was a fix. Is it valid now or not, I don't know. It could be that it's fixed in the meantime, don't know, I am fine with closing this, but it's more on your team to decide.

saledjenic commented:
Closing cause it's obsolete and seems not relevant anymore.
